cpp/src/arrow/status.h

```diff
@@ -95,7 +95,8 @@ enum class StatusCode : char {
   PythonError = 12,
   PlasmaObjectExists = 20,
   PlasmaObjectNonexistent = 21,
-  PlasmaStoreFull = 22
+  PlasmaStoreFull = 22,
+  CopyRequired = 23
```

Comment:
I don't think a new status code is needed. Can you return `Invalid` instead? As a matter of style, we should try not to use Status for routine error handling in C++. I think having this bubble up as an exception in Python is OK for now, but if we need to do zero-copy detection in C++, we are going to want to handle that in a different way.

Reply:
Done!